shotover-proxy/src/transforms/cassandra/sink_cluster/topology.rs

```diff
@@ -183,18 +226,147 @@ async fn fetch_current_nodes(
+                )),
+            }
+        } else {
+            Err(anyhow!("Failed to parse keyspace query response"))
+        }
+    }
+
+    pub fn build_keyspace(
+        mut row: Vec<MessageValue>,
+        data_center: &str,
+    ) -> Result<(String, KeyspaceMetadata)> {
+        let metadata = if let Some(MessageValue::Map(mut replication_strategy)) = row.pop() {
+            let strategy_name: String = match replication_strategy
+                .remove(&MessageValue::Varchar("class".into()))
+                .ok_or_else(|| anyhow!("replication strategy map should have a 'class' field",))?
+            {
+                MessageValue::Varchar(name) => name,
+                _ => return Err(anyhow!("'class' field should be a varchar")),
+            };
+
+            match strategy_name.as_str() {
+                "org.apache.cassandra.locator.SimpleStrategy" | "SimpleStrategy" => {
+                    let rf_str: String =
+                        match replication_strategy.remove(&MessageValue::Varchar("replication_factor".into())).ok_or_else(||
+                         anyhow!("SimpleStrategy in replication strategy map does not have a replication factor")
+                        )?{
+                            MessageValue::Varchar(rf) => rf,
+                            _ => return Err(anyhow!("SimpleStrategy replication factor should be a varchar "))
+                        };
+
+                    let replication_factor: usize = usize::from_str(&rf_str).map_err(|_| {
+                        anyhow!("Could not parse replication factor as an integer",)
+                    })?;
+
+                    KeyspaceMetadata { replication_factor }
+                }
+                "org.apache.cassandra.locator.NetworkTopologyStrategy"
+                | "NetworkTopologyStrategy" => {
+                    let data_center_rf = match replication_strategy
+                        .remove(&MessageValue::Varchar(data_center.into()))
+                    {
+                        Some(MessageValue::Varchar(rf_str)) => {
+                            usize::from_str(&rf_str).map_err(|_| {
+                                anyhow!("Could not parse replication factor as an integer",)
+                            })?
+                        }
+                        Some(_other) => {
+                            return Err(anyhow!(
+                                "NetworkTopologyStrategy replication factor should be a varchar"
+                            ))
+                        }
+                        None => 0,
+                    };
+
+                    KeyspaceMetadata {
+                        replication_factor: data_center_rf,
+                    }
+                }
+                "org.apache.cassandra.locator.LocalStrategy" | "LocalStrategy" => {
+                    KeyspaceMetadata {
+                        replication_factor: 0,
+                    }
+                }
+                _ => KeyspaceMetadata {
```

Comment:
If strategies are exhaustive we should return an Err here instead.
If they are not exhaustive because users can make custom ones or something, then we should log an error! here and probably return a replication_factor of 1 as that sounds like a safer default?